qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -154,6 +159,49 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
                 )
             )
             init.append(CommutativeCancellation())
+            # skip peephole optimization before routing if target basis gate set is discrete,
+            # i.e. only consists of Cliffords that an user might want to keep
+            # use rz, sx, x, cx as basis, rely on physical optimziation to fix everything later one
+            stdgates = get_standard_gate_name_mapping()
+
+            def _is_one_op_non_discrete(ops):
+                """Checks if one operation in `ops` is not discrete, i.e. is parameterizable
+                Args:
+                    ops (List(Operation)): list of operations to check
+                Returns
+                    True if at least one operation in `ops` is not discrete, False otherwise
+                """
+                for op in ops:
+                    if isinstance(op, str):
+                        op = stdgates.get(op, None)
+
+                    if (
+                        op is None
+                        or not isinstance(op, Instruction)
+                        or op.name in _discrete_skipped_ops
+                    ):
+                        continue
+
+                    if len(op.params) > 0:
+                        return True
+                return False
```

Comment:
We do have to use rz with multiples of `pi/2` for RB since the only singles in most IBM targets are id, sx, x, and rz. I don't see a reference to RB in #11354. What is the concern?

Reply:
We rolled back #11354 originally because it was causing issues with experiments. Since running synthesis before layout required us to not specify a target basis gate set and it was basically converting all 1q sequences of gates into `U` gates. So we quickly reverted it: https://github.com/Qiskit/qiskit/pull/11360 (and skipped an issue) to unblock experiments job against main. That's what this function and the if statement calling it are trying to guard against.

But now that I wrote it out, I'm not sure we actually need to guard against this with the way this implementation is being done. Since we're not actually synthesizing the unitaries until after layout and just collecting the blocks before layout, the synthesis should still target the correct basis and it probably would work as well as it did prior to this PR. So we probably can drop this conditional logic altogether because of the synthesis doesn't work with a discrete basis here then it wouldn't have worked before.

Reply:
Okay, so it sounds like we are okay, but I could try running the experiments tests with this PR if you want.

I looked up the details of the previous issue. It was a niche case. In the experiments tests, there is a test that sets the basis to `["h", "s", "x"]` and that test was failing because the conversion to parameterized gates couldn't be mapped back to the discrete basis (it seems like it should still be possible in principle?). In practice we usually run against IBM backends with rz.

At the time I noted that the following code would trigger the error:

```python
from qiskit import QuantumCircuit, transpile


circ = QuantumCircuit(1)
circ.s(0)
circ.y(0)

transpile(circ, basis_gates=["h", "s", "x"])
```

Reply:
This pass would not run with `basis_gates=["h", "s", "x"]`, so we are probably good to go? :-) Thanks @wshanks !

Reply:
Well I think the question is do we need to have this filter at all anymore? I'm thinking we might be ok just skipping this conditional logic but instead always run these 3 passes during init at level 2 and 3. It would be a potential issue if we were going to synthesize anything pre-layout we might avoid all the issues we had in the earlier attempt.

Reply:
It may be worthwhile to leave it in for any user that wants to play with Clifford circuits, right?

Reply:
Let's leave it for now then. But, my thinking was it wouldn't make a difference because we'll collect these gates post-routing as part of the peephole optimization we run during the optimization stage so filtering here doesn't make much of a difference.